llvm/lib/CodeGen/SelectionDAG/LegalizeDAG.cpp

```diff
@@ -5613,6 +5668,13 @@ void SelectionDAGLegalize::PromoteNode(SDNode *Node) {
                     DAG.getIntPtrConstant(0, dl, /*isTarget=*/true)));
     break;
   }
+  case ISD::VP_REDUCE_FADD:
+  case ISD::VP_REDUCE_FMUL:
+  case ISD::VP_REDUCE_FMAX:
+  case ISD::VP_REDUCE_FMIN:
+  case ISD::VP_REDUCE_SEQ_FADD:
```

Comment:
Why are `ISD::VP_REDUCE_ADD` and others not here? Are they handled elsewhere?

Reply:
I removed them because they weren't tested. The code was incorrect as written for min/max because it promoted using any_extend instead of sign/zero_extend. I can add it back and fix the bugs if you think I should. The function was introduced specifically for f16 on RISC-V.

Reply:
Thanks, I think it's OK to remove.